### PR TITLE
feat: `--debug.skip-fcu`

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Creates new `EngineApiSkipFcu` interceptor that skips `n` consecutive FCU messages if configured.

### CODE REVIEW

**Comment by onbjerg** [comment]
I assume this is to test behavior when FCUs are skipped? I wonder if we also should have one that randomly skips FCUs to sim unreliable connections

**Review by onbjerg — APPROVED** [review]

File: crates/node-builder/src/builder.rs [hunk 1]
```diff
@@ -665,6 +666,17 @@ where
...
+            debug!(target: "reth::cli", "spawning skip FCU task");
+            let (skip_fcu_tx, skip_fcu_rx) = unbounded_channel();
+            let engine_skip_fcu = EngineApiSkipFcu::new(skip_fcu_threshold);
+            executor.spawn_critical(
+                "skip FCU interceptor",
+                engine_skip_fcu.intercept(consensus_engine_rx, skip_fcu_tx),
+            );
+            consensus_engine_rx = skip_fcu_rx;
+        }
+
```
Comment (Rjected) [inline crates/node-builder/src/builder.rs:679]: should this be after the engine api store interceptor? in case you wanted to use both

**Review by Rjected — APPROVED** [review]
lgtm, the interceptors are pretty clean, just have one q

File: crates/node-builder/src/builder.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/node-builder/src/builder.rs:679]: the primary purpose of engine API store is replay. my thinking here is that you'd want to replay exactly the messages that reached the engine, alternatively we could add support for this arg in the replayer as well

**Pull request merged**